tests/virt_control_plane_test.go

```diff
@@ -0,0 +1,266 @@
+		countPods := 0
+		for _, pod := range podList.Items {
+			if pod.Status.Phase != expectedPhase {
+				continue
+			}
+			podReady := tests.PodReady(&pod)
+			if podReady != expectedConditionStatus {
+				continue
+			}
+			for _, podName := range podNames {
+				if strings.Contains(pod.Name, podName) {
+					if len(nodeNames) > 0 {
+						for _, nodeName := range nodeNames {
+							if pod.Spec.NodeName == nodeName {
+								countPods++
+							}
+						}
+					} else {
+						countPods++
+					}
+				}
+			}
+		}
+		return countPods
+	}
+
+	getPodList := func() (*v1.PodList, error) {
+		virtCli, err := kubecli.GetKubevirtClient()
+		if err != nil {
+			return nil, err
+		}
+		podList, err := virtCli.CoreV1().Pods("kubevirt").List(metav1.ListOptions{})
+		if err != nil {
+			return nil, err
+		}
+		return podList, nil
+	}
+
+	countReadyPodsOnNodes := func(podNames []string, nodeNames ...string) int {
+		podList, err := getPodList()
+		if err != nil {
+			return -1
+		}
+		return countPodsInStatus(podList, v1.PodRunning, v1.ConditionTrue, podNames, nodeNames...)
+	}
+
+	countPendingPods := func(podNames []string) int {
+		podList, err := getPodList()
+		if err != nil {
+			return -1
+		}
+		return countPodsInStatus(podList, v1.PodPending, v1.ConditionFalse, podNames)
+	}
+
+	runCommandOnNode := func(command string, node string, args ...string) (err error) {
+		cmdName := tests.GetK8sCmdClient()
+		newArgs := make([]string, 0)
+
+		// if the cluster is openshift we need to append `adm` for the commands `drain` and `uncordon`
+		// as the oc binary is used
+		if tests.IsOpenShift() {
+			if command == "drain" || command == "uncordon" {
+				newArgs = append(newArgs, "adm")
+			}
+		}
+
+		newArgs = append(newArgs, command)
+		newArgs = append(newArgs, node)
+		newArgs = append(newArgs, args...)
+		_, _, err = tests.RunCommandWithNS("", cmdName, newArgs...)
+		return
+	}
+
+	uncordonNode := func(node string) (err error) {
+		err = runCommandOnNode("uncordon", node)
+		return
+	}
+
+	uncordonNodes := func(nodeNames ...string) {
+		for _, nodeName := range nodeNames {
+			err = uncordonNode(nodeName)
+			tests.PanicOnError(err)
+		}
+	}
+
+	drainNodeWithTimeout := func(node string, podSelector string, timeoutAfterSeconds int) (err error) {
+		err = runCommandOnNode("drain", node, fmt.Sprintf("--timeout=%ds", timeoutAfterSeconds), podSelector)
+		return
+	}
+
+	waitForNodesToStabilize := func(expectedNumberOfReadyPods int, podNames []string, nodeNames ...string) {
+		Eventually(func() int { return countPendingPods(podNames) },
+			DefaultStabilizationTimeoutInSeconds, DefaultPollIntervalInSeconds,
+		).Should(Equal(0))
+		Eventually(func() int { return countReadyPodsOnNodes(podNames, nodeNames...) },
+			DefaultStabilizationTimeoutInSeconds, DefaultPollIntervalInSeconds,
+		).Should(BeNumerically(">=", expectedNumberOfReadyPods))
+	}
+
+	BeforeEach(func() {
+		tests.SkipIfNoCmd("kubectl")
+		tests.SkipIfSchedulableNodesLessThan(1)
```

Comment:
Just curious: under which circumstances can this happen?
Or should this be 2?

Reply:
Actually this is just a leftover. Will remove.